/*
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Library General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * E131ConfigMessages.proto
 * Defines the protocol buffers used to configure the Artnet device
 * Copyright (C) 2005 Simon Newton
 */

package ola.plugin.e131;

/*
 * Controls the preview bit for output ports.
 */
message PreviewModeRequest {
  required int32 port_id = 1;
  required bool preview_mode = 2;
  required bool input_port = 3;
}


message InputPortInfo {
  required int32 port_id = 1;
  required bool preview_mode = 2;
}


message OutputPortInfo {
  required int32 port_id = 1;
  required bool preview_mode = 2;
}


/*
 * Contains the status of input & output ports
 */
message PortInfoReply {
  repeated InputPortInfo input_port = 1;
  repeated OutputPortInfo output_port = 2;
}


/*
 * A generic request
 */
message Request {
  enum RequestType {
    E131_PORT_INFO = 1;
    E131_PREVIEW_MODE = 2;
  }

  required RequestType type = 1;
  optional PreviewModeRequest preview_mode = 2;
}

message Reply {
  enum ReplyType {
    E131_PORT_INFO = 1;
  }
  required ReplyType type = 1;
  optional PortInfoReply port_info = 2;
}
